netket/graph/lattice.py

```diff
@@ -261,6 +177,17 @@ def __init__(
                 to :math:`k`-nearest neighbor sites (measured by their Euclidean distance)
                 are included in the graph. The edges can be distiguished by their color,
                 which is set to :math:`k - 1` (so nearest-neighbor edges have color 0).
+                By default, nearest neighbours (:code:`max_neighbor_order=1`) are autogenerated
+                unless :code:`custom_edges` is passed.
+            custom_edges: Lists all edges starting in one unit cell, which are repeated
+                in every unit cell of the constructed lattice.
+                Should be a list of tuples; each tuple should contain the following:
+                    index of the starting point in the unit cell
+                    index of the endpoint in the unit cell
+                    vector pointing from the former to the latter
+                    color of the edge (optional)
+                If colors are not supplied, they are assigned sequentially starting from 0.
+                Cannot be used together with `max_neighbor_order`.
```

Comment:
ah. that's why you used None i guess.

what part of the logic would break down if both are specified?

Reply:
Nothing would stop you from specifying the same edge both ways or use colours in an inconsistent way.

Reply:
Basically @femtobit  and I thought that making both edge creation methods work nicely together would be too painful for the added convenience. If a user needs edges generated both ways, they can make two `Lattice`s, two `GraphOperator`s and then add those.

Reply:
ok... it's unfortunate but i see...

Maybe we should have two different methods in the future.
But for now this seems good.